sys/netinet/ip_carp.c

```diff
@@ -797,6 +804,25 @@ carp_prepare_ad(struct mbuf *m, struct carp_softc *sc, struct carp_header *ch)
  * To avoid LORs and possible recursions this function shouldn't
  * be called directly, but scheduled via taskqueue.
  */
+static void
+carp_reset_all(void *ctx __unused, int pending __unused)
+{
+	struct carp_softc *sc;
+
+	mtx_lock(&carp_mtx);
+	LIST_FOREACH(sc, &carp_list, sc_next)
+		if (sc->sc_state == MASTER) {
+			CARP_LOCK(sc);
+			CURVNET_SET(sc->sc_carpdev->if_vnet);
+			carp_set_state(sc, BACKUP, "resetting state");
+			carp_setrun(sc, 0);
+			carp_delroute(sc);
+			CURVNET_RESTORE();
+			CARP_UNLOCK(sc);
+		}
+	mtx_unlock(&carp_mtx);
+}
+
```

Comment:
V_carp_allow variable is per-VNET, while this code, being called by one of them, will affect all others too.